cli/tests/test_split_command.rs

```diff
@@ -675,3 +680,199 @@ fn test_split_interactive_with_paths() {
+    let secondary_path = test_env.env_root().join("secondary");
+
+    test_env.jj_cmd_ok(&main_path, &["desc", "-m", "first-commit"]);
+    std::fs::write(main_path.join("file1"), "foo").unwrap();
+    std::fs::write(main_path.join("file2"), "foo").unwrap();
+
+    // Create the second workspace and change its working copy commit to match
+    // the default workspace.
+    test_env.jj_cmd_ok(
+        &main_path,
+        &["workspace", "add", "--name", "second", "../secondary"],
+    );
+    // Change the working copy in the second workspace.
+    test_env.jj_cmd_ok(
+        &secondary_path,
+        &["edit", "-r", "description(first-commit)"],
+    );
+    let stdout = test_env.jj_cmd_success(&main_path, &["workspace", "list"]);
+    insta::assert_snapshot!(stdout, @r###"
+    default: qpvuntsm d12cb1c0 first-commit
+    second: qpvuntsm d12cb1c0 first-commit
+    "###);
+    // Check the working-copy commit in each workspace in the log output. The "@"
+    // node in the graph indicates the current workspace's working-copy commit.
+    insta::assert_snapshot!(get_workspace_log_output(&test_env, &main_path), @r###"
+    @  qpvuntsmwlqt default@ second@ first-commit
+    ◆  zzzzzzzzzzzz
+    "###);
+    insta::assert_snapshot!(get_workspace_log_output(&test_env, &secondary_path), @r###"
+    @  qpvuntsmwlqt default@ second@ first-commit
+    ◆  zzzzzzzzzzzz
+    "###);
+
+    // Do the split in the default workspace.
+    std::fs::write(
+        test_env.set_up_fake_editor(),
+        ["", "next invocation\n", "write\nsecond-commit"].join("\0"),
+    )
+    .unwrap();
+    test_env.jj_cmd_ok(&main_path, &["split", "file2"]);
+    // The working copy for both workspaces will be the second split commit.
+    let stdout = test_env.jj_cmd_success(&main_path, &["workspace", "list"]);
+    insta::assert_snapshot!(stdout, @r###"
+    default: vruxwmqv 97490ec0 second-commit
+    second: vruxwmqv 97490ec0 second-commit
+    "###);
+    insta::assert_snapshot!(get_workspace_log_output(&test_env, &main_path), @r###"
+    @  vruxwmqvtpmx default@ second@ second-commit
+    ○  qpvuntsmwlqt first-commit
+    ◆  zzzzzzzzzzzz
+    "###);
+    insta::assert_snapshot!(get_workspace_log_output(&test_env, &secondary_path), @r###"
+    @  vruxwmqvtpmx default@ second@ second-commit
+    ○  qpvuntsmwlqt first-commit
+    ◆  zzzzzzzzzzzz
+    "###);
+
+    // Test again with a --parallel split.
+    test_env.jj_cmd_ok(&main_path, &["undo"]);
+    std::fs::write(
+        test_env.set_up_fake_editor(),
+        ["", "next invocation\n", "write\nsecond-commit"].join("\0"),
+    )
+    .unwrap();
+    test_env.jj_cmd_ok(&main_path, &["split", "file2", "--parallel"]);
+    insta::assert_snapshot!(stdout, @r###"
+    default: vruxwmqv 97490ec0 second-commit
+    second: vruxwmqv 97490ec0 second-commit
+    "###);
+    insta::assert_snapshot!(get_workspace_log_output(&test_env, &main_path), @r###"
+    @  wqnwkozpkust default@ second@ second-commit
```

Comment:
Btw, i used `split --parallel` a day or two ago and wished it had left me on the first commit. Maybe we should change that behavior too. That's obviously out of scope for this PR. I just thought I'd mention it here.

Reply:
Yeah, I agree and had the same thought while I was working on this. I can update it in a follow up PR.